### DISCUSSION TITLE
Place your video script parts here

### DISCUSSION THREAD

**Discussion by MattHasson** [MFarabi619/hackateer#32] [General]
This is the discussion page to put your video script parts.

**Comment by MFarabi619** [comment]
Plan for the video, use the first minute to demo the project, and the last minute to go over infrastructure.

### First Draft:

**[00:00 - 00:10]**
(INTRO)
"Hello everyone, we're the team behind Hackateer, an innovative tool designed to streamline the brainstorming process for hackathon participants."

**[00:10 - 00:30]**
"Before we dive into the tool's features, let's quickly navigate through the user-friendly and fully responsive interface, complete with animated links and buttons. Let's start from the landing page."

**[00:30 - 00:50]**
(DESCRIPTION)
"Notice how each section, including the 'About Us' and 'Contact' pages, smoothly transitions from one to the other. You can learn more about our team and our mission on the About page, and get in touch with us through the Contact page."

**[00:50 - 01:10]**
(FEATURES)
"Now, onto the main functionality. Hackateer allows you to input problem statements, the hackathon's duration, and the specific skills of your team members."

**[01:10 - 01:30]**
"Based on these inputs, Hackateer generates high-quality, customized project suggestions. Each suggestion comes with a list of pros and cons, giving you a comprehensive understanding of potential outcomes."

**[01:30 - 01:50]**
(PROCESS)
"Behind the scenes, our team navigated through various stages - from concept discussion, design, front-end and back-end development, to cloud deployment."

**[01:50 - 02:00]**
(OUTRO)
"We're proud to introduce Hackateer - your partner in crafting winning hackathon ideas. Check it out and let us know what you think!"

**Comment by MFarabi619** [comment]
### Second Draft:

[00:00 - 00:10]
**(Intro)**
"Hello everyone, my name's Farabi. This is the project submission for the Hackateer team in the Hacks for Hackers Hackathon. The ideation stage of a hackathon is often the first hurdle that teams face. Our goal with Hackateer is to eliminate this pain point by not only providing custom-tailored ideas, but also their pros and cons along with a rough project timeline based on your team's size and skillset."

[00:10 - 00:30]
**(Show Hover and page loading animations)**
Hackateer has a well-designed, modern, and responsive user interface, complete with links and buttons. Each page smoothly transitions from one to the other. You can learn more about our team and our mission on the About page, and get in touch with us through the Contact page.

[00:30-00:50]
**(Show generate page functionality)**
Hackateer allows you to input problem statements, the hackathon's duration, and the specific skills of your team members. Hackateer generates high-quality, customized project suggestions. Each suggestion comes with a list of pros and cons, giving you a comprehensive understanding of the feasibility of each route.


[00:50-01:00]
**(Show Discord)**
Behind the scenes, our team navigated through various stages. We started on a discord group chat, and transitioned over to a discord server with dedicated channels. This ensured that our conversations were easy to track, and various topics were neatly separated and readily accessible. 

[01:00-1:10]
**(Show Figma)**
The application was designed on figma, starting with the color palette, typography, and branding. After which, the user-interface was wireframed, and then finalized at the design stage.

[01:10-1:20]
**(Show VS Code)**
The Front end was developed in Next.js, React, Tailwind CSS, and Typescript. It follows a component based architecture with a heavy priority on established best practices.

[01:20-1:30]
**(Show Github repository)**
For code collaboration, we used github. Each development stream has its own branch, and there's even a branch for documenting meeting minutes. 

[01:30-1:40]
**(Show Github Projects)**
For project management, we used Github projects. We linked issues to respective pull requests, and divided the project into manageable milestones with their unique issues. We also utilized labels, status, priority, and size metrics. 

[1:40-1:50]
**(Show Backend and Microsoft Azure)**
For the backend, a Node.js application is used to query the OpenAI API endpoint, and pass the fetched results to the frontend. The project is deployed to a Mircrosoft Azure Cloud Server. 

[1:50:-2:00]
**(Outro)**
We're very proud to introduce Hackateer - your partner in crafting winning hackathon ideas. Thank you very much for your time.
